raft/src/main/java/org/apache/kafka/raft/QuorumState.java

```diff
@@ -621,13 +621,14 @@ public void transitionToProspective() {
         }
 
         int retries = isCandidate() ? candidateStateOrThrow().retries() + 1 : 1;
+        int epoch = isResigned() ? epoch() + 1 : epoch();
```

Comment:
According to KIP-996, we should increase the epoch if the quorum state transitions from resigned.